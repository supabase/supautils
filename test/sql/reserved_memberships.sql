-- use a role that has the CREATEROLE privilege
set role rolecreator;
\echo

-- cannot grant reserved memberships
grant pg_read_server_files to anon;
grant pg_write_server_files to anon;
grant pg_execute_server_program to anon;
\echo

-- cannot grant reserved memberships when creating a role
create role tester in role pg_read_server_files;
create role tester in group anon, pg_write_server_files;
create role tester in role pg_execute_server_program, anon;
create role role_with_reserved_membership admin anon;
\echo

-- can grant non-reserved memberships
grant pg_monitor to fake;
grant pg_read_all_settings to fake;
grant pg_read_all_stats to fake;
\echo

-- can grant non-reserved memberships when creating a role
create role tester in role anon;
create role other admin anon;
