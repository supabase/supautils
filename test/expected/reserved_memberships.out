-- use a role that has the CREATEROLE privilege
set role rolecreator;
\echo

-- cannot grant reserved memberships
grant pg_read_server_files to anon;
ERROR:  "pg_read_server_files" role memberships are reserved, only superusers can grant them
grant pg_write_server_files to anon;
ERROR:  "pg_write_server_files" role memberships are reserved, only superusers can grant them
grant pg_execute_server_program to anon;
ERROR:  "pg_execute_server_program" role memberships are reserved, only superusers can grant them
\echo

-- cannot grant reserved memberships when creating a role
create role tester in role pg_read_server_files;
ERROR:  "pg_read_server_files" role memberships are reserved, only superusers can grant them
create role tester in group anon, pg_write_server_files;
ERROR:  "pg_write_server_files" role memberships are reserved, only superusers can grant them
create role tester in role pg_execute_server_program, anon;
ERROR:  "pg_execute_server_program" role memberships are reserved, only superusers can grant them
create role role_with_reserved_membership admin anon;
ERROR:  "role_with_reserved_membership" role memberships are reserved, only superusers can grant them
\echo

-- can grant non-reserved memberships
grant pg_monitor to fake;
grant pg_read_all_settings to fake;
grant pg_read_all_stats to fake;
\echo

-- can grant non-reserved memberships when creating a role
create role tester in role anon;
create role other admin anon;
